### PR TITLE
ENH: expose test runner raise_warnings option

### PR DESCRIPTION
**Pull request by mattip**
my first pull request, useful to turn Deprecated errors into warnings

### CODE REVIEW

File: runtests.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ def main(argv):
     parser.add_argument("--bench-compare", action="store", metavar="COMMIT",
                         help=("Compare benchmark results to COMMIT. "
                               "Note that you need to commit your changes first!"))
+    parser.add_argument("--raise_warnings", default=None, type=str,
```
Comment (njsmith) [inline runtests.py]: Should use `-` instead of `_` for consistency with other flags.

**Comment by njsmith** [comment]
Otherwise lgtm

**Comment by charris** [comment]
Might want to squash the commits and follow the commit message guidelines in `doc/source/dev/gitwash/development_workflow.rst`. Do `git rebase -i HEAD^^` and follow directions to edit the last two commits, then do a force push to origin, `git push -f origin <your branch>`.

**Comment by charris** [comment]
Thanks @mattip .

**Pull request merged**